timemachine/cpp/src/nonbonded_all_pairs.cu

```diff
@@ -376,26 +407,31 @@ void NonbondedAllPairs<RealType, Interpolated>::execute_device(
 
     // coords are N,3
     if (d_du_dx) {
-        k_inv_permute_accum<<<dimGrid, tpb, 0, stream>>>(N, d_perm_, d_sorted_du_dx_, d_du_dx);
+        k_scatter_accum<<<dim3(ceil_divide(K_, tpb), 3, 1), tpb, 0, stream>>>(
```

Comment:
If I understand the `scatter gather` paradigm, scatter and accumulate are antonyms?

Reply:
Oh, maybe this is confusing.. my intent was to generalize the naming when going from the existing K=N case to the more general K<=N as:
- permute -> scatter
- inverse permute -> gather

This is intended to be independent of whether we accumulate or assign to the result array. Does that make sense?

Reply:
I am more familiar with the `fork and join` paradigm and it seemed like fork and scatter are equivalent and gather and join are equivalent. In which case this seemed backwards? I might just not understand the paradigm, was unexpected to see a scatter (or what I interpreted as a broadcast/fork) to reduce.  

Doesn't matter as long as it is a consistent convention

Reply:
>The picture I had for "scatter" was https://pytorch-scatter.readthedocs.io/en/latest/functions/scatter.html .

Err, posted that before I saw @mcwitt 's link https://github.com/proteneer/timemachine/pull/660#discussion_r826247997 , didn't mean to override that. (I think the only difference is the wiki definition and the current function assume no repeated scatter idxs, while the pytorch_scatter reduces over any repeated scatter idxs. For the current use, only unique idxs make sense.)

> I might just not understand the paradigm, was unexpected to see a scatter (or what I interpreted as a broadcast/fork) to reduce.

I had misread the `y[idxs[i]] += x[i]` as a reduction, but @mcwitt clarified offline that it's always a single addition to whatever was in the output array before.